source/agora/common/DNS.d

```diff
@@ -136,6 +177,45 @@ public struct RDATA
     public uint address;
 }
 
+/// https://datatracker.ietf.org/doc/html/rfc3596#section-2.2
+public struct AAAARDATA
+{
+    /// A 128 bit IPv6 address is encoded in the data portion of an AAAA
+    /// resource record in network byte order (high-order byte first).
+    public ulong[2] address;
+}
+
+/// https://datatracker.ietf.org/doc/html/rfc2782
+public struct SRVRDATA
+{
+    /// The priority of this target host (lowest is better)
+    public ushort priority;
+
+    /// A server selection mechanism for entries with the same priority
+    public ushort weight;
+
+    // The port on this target host of this service.
+    public ushort port;
+
+    /// The domain name of the target host (should not be an alias)
+    /// Using '.' means that the service doesn't exists.
+    public Domain target;
+}
+
+/// https://datatracker.ietf.org/doc/html/rfc7553#section-4
+public struct URIRDATA
+{
+    /// The priority of this target host (lowest is better)
+    public ushort priority;
+
+    /// A server selection mechanism for entries with the same priority
+    public ushort weight;
+
+    /// This field holds the URI of the target (RFC 3986)
+    /// Resolution of the URI is according to the definitions for the Scheme of the URI.
+    public string target;
```

Comment:
`Address` can be used here

Reply:
Yes it could, but I think it's better not to do it here. This module is currently almost standalone: The only two dependencies, outside of stdlib, are `Ensure` (itself a leaf module) and `Serialization` (a library).
There is a possibility that we will later move this to its own library / somewhere else, so let's keep it as a string and have Agora validate it independently for the time being.